### PR TITLE
EOS-27648 Adding config updates for RGW and marking deleted items

### PR DESCRIPTION
**Pull request by zahidsk**
Signed-off-by: Zahid Shaikh <zahid.shaikh@seagate.com>

### CODE REVIEW

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml [hunk 1]
```diff
@@ -5,15 +5,15 @@ cortx:
...
-    openldap:
-      endpoints:
-        - ldap://<<.Values.cortx.external.openldap.endpoints>>:389
-        - ssl://<<.Values.cortx.external.openldap.endpoints>>:636
-      servers:
-        <<.Values.cortx.external.openldap.servers>>
-      admin: admin
-      secret: openldap_admin_secret
-      base_dn: dc=seagate,dc=com
+    openldap:                                                          #Deleted
```
Comment (tanujashinde0405) [inline k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml:8]: Confirm from @kailashyadav11 if csm is getting deployed without openldap so we can remove those entires as pat of kailash ticket.

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml [hunk 2]
```diff
@@ -5,15 +5,15 @@ cortx:
...
+      endpoints:                                                       #Deleted
```
Comment (ujjwalpl) [inline k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml:9]: Use Deprecated to be more precise.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml [hunk 3]
```diff
@@ -44,22 +44,22 @@ cortx:
     auth_user: user_name
     auth_admin: sgiamadmin
     auth_secret: s3_auth_admin_secret
-    iam:
-      endpoints:
-      - https://<<.Values.cortx.io.svc>>:9443
+    iam:                                                               #Deleted
```
Comment (ujjwalpl) [inline k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml:47]: rgw is new section hence nothing can be deleted. Mention a separate section named s3

```
s3:                                                       # deprecated. 
  iam:                                                    # deprecated. 
...
```

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml (hunk 1 above)
Comment (ujjwalpl) [reply k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml:8]: @tanujashinde0405  Lets wait. As per protocol, entries goto deprecated first.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml (hunk 1 above)
Comment (tanujashinde0405) [reply k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml:8]: Okay sure, We had a ticket to remove this entry: https://jts.seagate.com/browse/EOS-28037, But we can mark them as deprecated also.

**Comment by nitin-seagate** [comment]
My comments are in-line with Ujjwal...

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml (hunk 1 above)
Comment (zahidsk) [reply k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml]: Fixed, changed to deprecated

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml (hunk 3 above)
Comment (zahidsk) [reply k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-configmap/templates/config-template.yaml]: added a s3 section and marked it as deprecated

**Review by ujjwalpl — APPROVED** [review]

**Pull request merged**